core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -848,15 +855,81 @@ public Graph build() {
         }
 
         private void updateGraphHooks(final GraphConfig config) {
-            List<GraphHook> hooks = config.getHooks();
-            if (store.isSupported(AddNamedView.class) && !hasHook(hooks, NamedViewResolver.class)) {
-                hooks.add(0, new NamedViewResolver(config.getGraphId()));
-            }
-            if (store.isSupported(AddNamedOperation.class) && !hasHook(hooks, NamedOperationResolver.class)) {
-                config.getHooks().add(0, new NamedOperationResolver(config.getGraphId()));
-            }
+            updateNamedViewResolverHook(config);
+
+            updateNamedOperationResolverHook(config);
+
+            updateFunctionAuthoriserHook(config);
+        }
+
+        private void updateFunctionAuthoriserHook(final GraphConfig config) {
+            final List<GraphHook> hooks = config.getHooks();
             if (!hasHook(hooks, FunctionAuthoriser.class)) {
-                config.getHooks().add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+                LOGGER.warn("No FunctionAuthoriser hook was supplied, adding default hook.");
+                hooks.add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+            }
+        }
+
+        private void updateNamedViewResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedView.class,
+                    NamedViewResolver.class,
+                    null != properties ? properties.getCacheServiceNamedViewSuffix(config.getGraphId()) : null);
+        }
+
+        private void updateNamedOperationResolverHook(final GraphConfig config) {
+            updateGetFromCacheHookWhenMissingOrHasWrongSuffix(
+                    config,
+                    AddNamedOperation.class,
+                    NamedOperationResolver.class,
+                    null != properties ? properties.getCacheServiceNamedOperationSuffix(config.getGraphId()) : null);
+        }
+
+        /**
+         * This method is designed to keep suffix used for cache names aligned between
+         * a Handler which Adds to the cache and ResolverHook which Gets from the cache.
+         * If the suffix is mismatched then writing and reading to cache will not behave correctly.
+         *
+         * @param config           the graphConfig containing the Hooks
+         * @param operationClass   the Operation requiring cache write
+         * @param hookClass        the Hook requiring cache reading
+         * @param suffixFromProperties the suffix from property
+         */
+        private void updateGetFromCacheHookWhenMissingOrHasWrongSuffix(final GraphConfig config, final Class<? extends Operation> operationClass, final Class<? extends GetFromCacheHook> hookClass, final String suffixFromProperties) {
```

Comment:
Within UpdateHooks
Update..GetFromCacheHook... When Missing or HasWrongSuffix.
So when the GetFromCacheHook is missing it updates Hooks add in the required hook.
If not missing it checks if its correct.

I guess it could be called `validateGetFromCacheHookForMissingOrHasWrongSuffix`

Reply:
`validateGetFromCacheHook` please